MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -6027,8 +5959,19 @@ public void changeStatus(Person person, int status) {
             // Don't forget to tell the spouse
             if (person.hasSpouse()) {
                 Person spouse = person.getSpouse();
+
+                spouse.addFormerSpouse(new FormerSpouse(person.getId(),
+                        FormerSpouse.convertDateTimeToLocalDate(getDateTime()), FormerSpouse.REASON_WIDOWED));
+                person.addFormerSpouse(new FormerSpouse(spouse.getId(),
+                        FormerSpouse.convertDateTimeToLocalDate(getDateTime()), FormerSpouse.REASON_WIDOWED));
+
+                if (!getCampaignOptions().getKeepMarriedNameUponSpouseDeath() && (spouse.getMaidenName() != null)) {
+                    spouse.setSurname(spouse.getMaidenName());
+                }
+
                 PersonalLogger.spouseKia(spouse, person, getDate());
-                spouse.setSpouseID(null);
+                spouse.setSpouseId(null);
+                spouse.setMaidenName(null);
             }
```

Comment:
I'll take a look at moving this into Person later tonight, as it really should just use the divorce method.

Reply:
Moved and fixed a bug in the divorce code.